osbuild/main_cli.py

```diff
@@ -149,41 +149,59 @@ def osbuild_cli():
         monitor_name = "NullMonitor" if args.json else "LogMonitor"
     monitor = osbuild.monitor.make(monitor_name, args.monitor_fd)
 
+    build_result = None
     try:
         with ObjectStore(args.store) as object_store:
             stage_timeout = args.stage_timeout
 
             pipelines = manifest.depsolve(object_store, exports)
 
-            manifest.download(object_store, monitor, args.libdir)
+            download_results = manifest.download(
+                    object_store,
+                    monitor,
+                    args.libdir)
 
-            r = manifest.build(
-                object_store,
-                pipelines,
-                monitor,
-                args.libdir,
-                stage_timeout=stage_timeout
-            )
+            download_sucess = True
+            for download_result in download_results:
+               download_sucess = download_result["success"] and download_sucess
+
+            if download_sucess:
+                build_result = manifest.build(
+                    object_store,
+                    pipelines,
+                    monitor,
+                    args.libdir,
+                    stage_timeout=stage_timeout
+                )
+
+                if build_result["success"] and exports:
+                    for pid in exports:
+                        export(pid, output_directory, object_store, manifest)
+            else:
+                monitor.log("Downloading failed")
 
-            if r["success"] and exports:
-                for pid in exports:
-                    export(pid, output_directory, object_store, manifest)
 
     except KeyboardInterrupt:
         print()
         print(f"{RESET}{BOLD}{RED}Aborted{RESET}")
         return 130
 
     if args.json:
-        r = fmt.output(manifest, r)
-        json.dump(r, sys.stdout)
-        sys.stdout.write("\n")
+        if build_result:
+            build_result = fmt.output(manifest, build_result)
+            json.dump(build_result, sys.stdout)
+            sys.stdout.write("\n")
+        else:
+            # TODO this will make composer fail at unmarshalling the json and
```

Comment:
Would this be a regression, or do we anyway return invalid json at the moment if downloading failed?

Reply:
So far no I don't think so, because if a download fails osbuild crashes in the current version. So I think this bit would let the old behavior occur for unknown exceptions.